tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js

```diff
@@ -0,0 +1,548 @@
+        delAndAssertResult(s3, { bucket, key, versionId,
+            resultType: deleteDeleteMarker }, next);
+    }, () => cb());
+}
+
+function _awsGetLatestVerId(key, body, cb) {
+    return awsS3.getObject({ Bucket: awsBucket, Key: key }, (err, result) => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `getting object from AWS, got error ${err}`);
+        const resultMD5 = expectedETag(result.Body, false);
+        const expectedMD5 = expectedETag(body, false);
+        assert.strictEqual(resultMD5, expectedMD5);
+        return cb(null, result.VersionId);
+    });
+}
+
+function _getAssertDeleted(s3, params, cb) {
+    const { key, versionId, errorCode } = params;
+    return s3.getObject({ Bucket: bucket, Key: key, VersionId: versionId },
+        err => {
+            assert.strictEqual(err.code, errorCode);
+            assert.strictEqual(err.statusCode, 404);
+            return cb();
+        });
+}
+
+function _awsGetAssertDeleted(params, cb) {
+    const { key, versionId, errorCode } = params;
+    return awsS3.getObject({ Bucket: awsBucket, Key: key, VersionId:
+        versionId }, err => {
+        assert.strictEqual(err.code, errorCode);
+        assert.strictEqual(err.statusCode, 404);
+        return cb();
+    });
+}
+
+describeSkipIfNotMultiple('AWS backend delete object w. versioning',
+function testSuite() {
+    this.timeout(30000);
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+        beforeEach(() => {
+            process.stdout.write('Creating bucket\n');
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
+            .catch(err => {
+                process.stdout.write(`Error creating bucket: ${err}\n`);
+                throw err;
+            });
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write('Error emptying/deleting bucket: ' +
+                `${err}\n`);
+                throw err;
+            });
+        });
+
+        it('versioning not configured: should delete a specific version ' +
+        'in AWS backend successfully', done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putToAwsBackend(s3, bucket, key, someBody,
+                    err => next(err)),
+                next => _awsGetLatestVerId(key, someBody, next),
+                (awsVerId, next) => delAndAssertResult(s3, { bucket,
+                    key, versionId: 'null', resultType: deleteVersion },
+                    err => next(err, awsVerId)),
+                (awsVerId, next) => _awsGetAssertDeleted({ key,
```

Comment:
why is it actually deleting it? we don't want to delete anything on real AWS unless they have versioning enabled on our s3 and they request to delete a specific version. otherwise, it should be putting a delete marker on aws.  As discussed, it's ok that deleting since specifying null.  We should add a test where we specify an actual version id other than null and confirm that the AWS version is left alone.